components/style/values/specified/image.rs

```diff
@@ -239,9 +238,20 @@ impl ToCss for AngleOrCorner {
             AngleOrCorner::Angle(angle) => angle.to_css(dest),
             AngleOrCorner::Corner(horizontal, vertical) => {
                 try!(dest.write_str("to "));
-                try!(horizontal.to_css(dest));
-                try!(dest.write_str(" "));
-                try!(vertical.to_css(dest));
+                match (horizontal, vertical) {
+                    (Some(horizontal), Some(vertical)) => {
+                        try!(horizontal.to_css(dest));
+                        try!(dest.write_str(" "));
+                        try!(vertical.to_css(dest));
+                    }
+                    (Some(horizontal), None) => {
+                        try!(horizontal.to_css(dest));
+                    }
+                    (None, Some(vertical)) => {
+                        try!(vertical.to_css(dest));
+                    }
+                    (None, None) => unreachable!()
+                }
```

Comment:
You can restructure this with match something like this:
```rust
let mut horizontal_present = false;
if let Some(horizontal) = horizontal {
    try!(horizontal.to_css(dest));
    horizontal_present = true;
}
if let Some(vertical) = vertical {
    if horizontal_present {
        try!(dest.write_str(" "));
    }
    try!(vertical.to_css(dest));
}
```
This looks a bit clearer.